### PR TITLE
Added assertArrayIsList()

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

**Comment by kukulich** [comment]
@sebastianbergmann Would you accept this when I add more tests etc?

**Comment by sebastianbergmann** [comment]
> @sebastianbergmann Would you accept this when I add more tests etc?

Yes.

File: src/Framework/Assert.php [hunk 1]
```diff
@@ -113,6 +114,15 @@ public static function assertArrayNotHasKey(int|string $key, array|ArrayAccess $
         static::assertThat($array, $constraint, $message);
     }
 
+    public static function assertArrayIsList(array $array, string $message = ''): void
+    {
+        static::assertThat(
+            $array,
+            new ArrayIsList(),
```
Comment (sebastianbergmann) [inline src/Framework/Assert.php:121]: Please no superfluous `()` when no constructor arguments are passed.

**Comment by kukulich** [comment]
It looks the fail of type checker is not related to this patch.

**Comment by sebastianbergmann** [comment]
> It looks the fail of type checker is not related to this patch.

I'll look into this.

File: src/Framework/Constraint/Traversable/ArrayIsList.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    protected function matches(mixed $other): bool
+    {
+        if (!is_array($other)) {
+            return false;
+        }
+
+        if ($other === []) {
+            return true;
+        }
+
```
Comment (derrabus) [inline src/Framework/Constraint/Traversable/ArrayIsList.php:40]:
```suggestion

        if (function_exists('array_is_list')) {
            return array_is_list($other);
        }

```

On PHP 8.1, we have [a native function](https://www.php.net/array_is_list) for that. I suggest we leverage it if it's available.

**Comment by sebastianbergmann** [comment]
Merged manually. Thanks!

**Pull request closed without merging**